.github/workflows/verify.yml

```diff
@@ -0,0 +1,38 @@
+# This workflow will build a Java project with Maven, and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-java-with-maven
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: Java CI with Maven
+
+on: push
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Source
+        uses: actions/checkout@v3
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'temurin'
+          cache: maven
+
+      - name: Compile Sources
+        run: mvn -B compile --file pom.xml
+
+      - name: Test
+        run: mvn -B test --file pom.xml
+
+      - name: Package
+        run: mvn -B package --file pom.xml -DskipTests=true
```

Comment:
I had to realise that maven is not so good at determining what it has already done and what it has not. Without this parameter (-DskipTests=true), the tests are executed not once but three times (test, package and verify).